projects/storefrontlib/router/on-navigate.service.ts

```diff
@@ -85,17 +85,49 @@ export class OnNavigateService {
               return;
             }
 
-            setTimeout(
-              () => this.viewportScroller.scrollToPosition([0, 0]),
-              100
-            );
+            this.scrollToPosition(currentRoute, position);
           }
 
           this.hostComponent?.location?.nativeElement.focus();
         });
     }
   }
 
+  /**
+   * Scrolls to a specified position or anchor based on the current route and configuration.
+   * @param currentRoute The current route containing scroll information.
+   * @param position The target scroll position as [x, y] coordinates, or null.
+   */
+  private scrollToPosition(
+    currentRoute: Scroll,
+    position: [number, number] | null
+  ): void {
+    const scrollTo = (
+      anchor: string | null,
+      scrollPosition: [number, number]
+    ) => {
+      if (
+        anchor &&
+        (this.router as any).options?.anchorScrolling === 'enabled'
```

Comment:
This is using private API of Angular Router. See Angular source code:
https://github.com/angular/angular/blob/1e3bcfecbab260fb66a1ac50e77192bb98cc0f3e/packages/router/src/router.ts#L83

Private API might be changed anytime by Angular team (even in patch releases), therefore by using this API we're risking that potentially our library code may break in runtime anytime when customer updates their Angular version to a newer one, that changes the name of this property `options`. To be honest, I believe there are small chances Angular team will rename this property out of sudden, but they have right to do it. So _in general_ we should avoid using private API of Angular.

Fortunately the injection token `ROUTER_CONFIGURATION` is public, so we can safely inject it in our class and use it. So instead of picking private property `options` from `Router`, let's inject the `ROUTER_CONFIGURATION` ourselves, e.g.
```ts
protected readonly routerConfiguration  = inject(ROUTER_CONFIGURATION, {optional: true}) || {};
```
```ts
...
if (
  ... &&
  this.routerConfiguration.anchorScrolling === 'enabled'
```